### PR TITLE
Support direct posting store API for single string attribute.

### PR DESCRIPTION
**Pull request by geirst**
@baldersheim please review
@toregge FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/singlestringpostattribute.hpp [hunk 1]
```diff
@@ -152,5 +154,12 @@ SingleValueStringPostingAttributeT<B>::getSearch(QueryTermSimpleUP qTerm,
                                 *this);
 }
 
-} // namespace search
+template <typename B>
```
Comment (baldersheim) [inline searchlib/src/vespa/searchlib/attribute/singlestringpostattribute.hpp:157]: This is so simple you can just have it in the header file.

**Review by baldersheim — DISMISSED** [review]
2 comments

File: searchlib/src/tests/attribute/direct_multi_term_blueprint/direct_multi_term_blueprint_test.cpp [hunk 2]
```diff
@@ -19,13 +20,22 @@ using namespace search::queryeval;
...
+using LookupKey = IDirectPostingStore::LookupKey;
+
+struct IntegerKey : public LookupKey {
     int64_t _value;
     IntegerKey(int64_t value_in) : _value(value_in) {}
     vespalib::stringref asString() const override { abort(); }
     bool asInteger(int64_t& value) const override { value = _value; return true; }
 };
 
+struct StringKey : public IDirectPostingStore::LookupKey {
```
[the remark this answers is not in the document]
Comment (geirst) [reply searchlib/src/tests/attribute/direct_multi_term_blueprint/direct_multi_term_blueprint_test.cpp]: Fixed

File: searchlib/src/vespa/searchlib/attribute/singlestringpostattribute.hpp (hunk 1 above)
Comment (geirst) [reply searchlib/src/vespa/searchlib/attribute/singlestringpostattribute.hpp]: Fixed

**Comment by geirst** [comment]
@baldersheim PTAL

**Review by baldersheim — APPROVED** [review]

**Pull request merged**